### ISSUE TITLE
tag functions from std and other crates?

### ISSUE THREAD

**Issue by dckc** [thepowersgang/tag_safe#1]
I'd like to use this `tag_unsafe` lint to enforce object capability discipline. To do this, I need to be able to tag existing functions from std and other crates... a sort of  [taming database](http://erights.org/elib/legacy/index.html).

Does that seem feasible?

I'm willing to code up the feature, given a little guidance.

**Comment by thepowersgang** [comment]
That would be a good idea - I'm starting on a redesign of this crate to properly specify its behaviour (and hopefully include external crate handling).

I'm not sure how easy an external database will be, but hopefully it shouldn't be too difficult.

**Comment by dckc** [comment]
"database" might give the wrong impression... a csv or .toml file would be fine. And it need not be dynamic. Baking it in at build time would work.

**Comment by dckc** [comment]
I think my preferred form of database might be rust source:

```rust
#[tag_unsafe(ocap)]
use std::fs::File::open;
```

**Comment by thepowersgang** [comment]
I've drafted up a rewrite of the plugin (See the [redesign branch](/thepowersgang/tag_safe/browse/redesign)) that both cleans up some of the logic, and can parse a list of safe/unsafe functions from a linked file. (Just a simple list of function names as pretty-printed by the compiler)

```rust
#[tagged_safe(ocap="tagsafe_std_ocap.txt")]
extern crate std as _std;
```

**Comment by dckc** [comment]
I see this was fix Nov 25, 2016 in 09c2c25
with documentation in d66a166b7f2cafa74f52d9c88bfe6628da2ee572
